lib/rucio/tests/test_tpc.py

```diff
@@ -0,0 +1,168 @@
+# -*- coding: utf-8 -*-
+# Copyright 2012-2021 CERN
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Authors:
+# - Mayank Sharma <mayank.sharma@cern.ch>, 2021
+
+import datetime
+import pytest
+import hashlib
+import re
+
+from rucio.core.rule import add_rule
+from rucio.core.transfer import get_transfer_requests_and_source_replicas
+from rucio.common.utils import generate_uuid
+from rucio.daemons.judge.evaluator import re_evaluator
+from rucio.daemons.conveyor import submitter, poller, finisher
+from rucio.client.rseclient import RSEClient
+from rucio.client.ruleclient import RuleClient
+from rucio.common.utils import run_cmd_process
+
+# Fixtures
+
+
+@pytest.fixture
+def did_factory(vo, test_scope):
+    from rucio.tests.temp_factories import TemporaryDidFactory
+
+    with TemporaryDidFactory(vo=vo, default_scope=test_scope) as factory:
+        yield factory
+
+
+@pytest.fixture
+def rse_client():
+    return RSEClient()
+
+
+@pytest.fixture
+def rule_client():
+    return RuleClient()
+
+
+@pytest.fixture
+def rse1(containerized_rses):
+    if len(containerized_rses) > 1:
+        rse, rse_id = containerized_rses[0]
+        return {
+            'rse_name': rse,
+            'rse_id': rse_id,
+        }
+    return None
+
+
+@pytest.fixture
+def rse2(containerized_rses):
```

Comment:
Aye! I've removed these fixtures for now. I wanted to create tuples of rse_pairs and parametrize the test_tpc with it. It was a bit complicated since fixtures (containerized_rses) cannot be called as regular functions in non-test functions and also pytest.mark.parameterize does not accept return values from fixtures. For the time being, rse_1 and rse_2 were placeholders for the values that would have been passed. I will create a separate PR for this.